core/src/main/java/org/apache/calcite/rex/RexLiteral.java

```diff
@@ -576,6 +579,35 @@ private String intervalString(BigDecimal v) {
     return b.toString();
   }
 
+  private static BigDecimal getMultiplier(TimeUnit unit) {
```

Comment:
So, is there a reason you did this instead of changing the multiplier values here: `org/apache/calcite/avatica/util/TimeUnit.java`

Reply:
This multiplier values need to be increased by 10^6, and the multiplier of subsecond units are all 1 in `TimeUnit.java`

Reply:
this file is not in calcite repo